File: web/src/lib/api/profileService.ts
import { apiGet, apiPost, apiPut } from './client';

export const profileService = {
  getProfile: () => apiGet<any>('/api/v1/business-profile/profile'),
  updateProfile: (updates: any) => apiPut('/api/v1/business-profile/profile', updates),
  getTenant: () => apiGet<any>('/api/v1/business-profile/tenant'),
  updateTenant: (updates: any) => apiPut('/api/v1/business-profile/tenant', updates),
  getStaff: () => apiGet('/api/v1/business-profile/staff'),
  inviteStaff: (email: string, role: string) => apiPost('/api/v1/business-profile/staff', { email, role }),
  deleteAccount: () => apiPost('/api/v1/business-profile/account/delete'),
};

File: tumbuhin_flutter/lib/shared/services/profile_service.dart
import '../models/user_profile.dart';
import '../models/tenant.dart';
import '../../../core/api/api_client.dart';

class ProfileService {
  final ApiClient _apiClient;

  ProfileService(this._apiClient);

  Future<UserProfile> getProfile() async {
    final response = await _apiClient.dio.get(
      '/api/v1/business-profile/profile',
    );
    return UserProfile.fromJson(response.data);
  }

  Future<Tenant> getTenant() async {
    final response = await _apiClient.dio.get(
      '/api/v1/business-profile/tenant',
    );
    return Tenant.fromJson(response.data);
  }

  Future<UserProfile> updateProfile(Map<String, dynamic> updates) async {
    final response = await _apiClient.dio.put(
      '/api/v1/business-profile/profile',
      data: updates,
    );
    return UserProfile.fromJson(response.data);
  }

  Future<Tenant> updateTenant(Map<String, dynamic> updates) async {
    final response = await _apiClient.dio.put(
      '/api/v1/business-profile/tenant',
      data: updates,
    );
    return Tenant.fromJson(response.data);
  }
}
